Training24/Test.c

```diff
@@ -1,19 +1,141 @@
-// -------------------------------------
-// Standard Test file in Main
+// Dhaarani Mohanasundaram
 // -------------------------------------
 // Test.c
-// Program on main branch.
+// Program on A5 branch.
+// Implements test cases and checks the user input for sort & search in integer array.
 // ------------------------------------------------------------------------------------
 #include <stdio.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
 #include "Header.h"
 
-int main() {
-const char* message = GetMessage ();
-printf ("%s\n", message);
-return;
+#define RED "\x1b[31m"
+#define GREEN "\x1b[32m"
+#define BLUE "\x1b[34m"
+#define MAGENTA "\x1b[35m"
+#define CYAN "\x1b[36m"
+#define RESET "\x1b[0m"
+#define MAX_SIZE 100
+
+/// <summary>Runs few testcases automatically.</summary>
+void TestCases ();
+
+/// <summary>Sorts and search for user input.</summary>
+void SortAndSearchUserInput ();
+
+int main () {
+   char input[MAX_SIZE];
+   while (1) {
+      printf ("\nChoose an option:\n"
+         "1. Run Test Cases\n"
+         "2. Enter User Input\n"
+         "3. Exit\n"
+         "\nEnter your choice (1-3): ");
+      if (fgets (input, sizeof (input), stdin) != NULL) {
+         int len = (int)strlen (input);
+         if (len > 0 && input[len - 1] != '\n') while (getchar () != '\n');
+      }
+      char* endPtr;
+      int choice = strtol (input, &endPtr, 10);
+      if (*endPtr != '\0' && !isspace (*endPtr)) {
+         printf ("Invalid choice. Please enter a valid number (1, 2, or 3).\n");
+         continue;
+      }
+      switch (choice) {
+      case 1: TestCases (); break;
+      case 2: SortAndSearchUserInput (); break;
+      case 3: printf (MAGENTA "\n\tEXITING THE PROGRAM.....!!\n" RESET); return 0;
+      default: printf ("Invalid choice. Please enter 1 or 2 or 3.\n"); break;
+      }
+   }
+}
+
+void TestCases () {
+   int testCases[][MAX_SIZE] = { {1, 2, 3, 4, 5, 6, 7, 8}, {0, -3, 1, 3, 5, 7, -9}, {10, -20, -30, 40, 0},
+                                 {3, -3, 3, -3, -3, -3}, {5, 5, 5, 5, 5, 5, 5, 5} },
+      expectedSorted[][MAX_SIZE] = { {1, 2, 3, 4, 5, 6, 7, 8}, {-9, -3, 0, 1, 3, 5, 7}, {-30, -20, 0, 10, 40},
+                                     {-3, -3, -3, -3, 3, 3}, {5, 5, 5, 5, 5, 5, 5, 5} },
+      expectedIndex[][2] = { {4, 3}, {7, 6}, {20, -1}, {3, 4}, {5, 0} },
+      sizes[] = { 8, 7, 5, 6, 8 },
+      sortedArr[MAX_SIZE] = { 0 };
+   printf (CYAN "\n\t       -----------------    TEST CASES FOR SORTING AND SEARCHING    -----------------\n" RESET
+      "\n\t+-----------------------+-----------------------+----------------+--------------+--------+\n"
+      "\t|    UNSORTED  ARRAY    |     SORTED  ARRAY     | ELEMENT SEARCH | ACTUAL INDEX | RESULT |\n"
+      "\t+-----------------------+-----------------------+----------------+--------------+--------+\n");
+   for (int i = 0; i < 5; i++) {
+      int n = sizes[i], searchElement = expectedIndex[i][0], k = 0;
```

Comment:
Please rename this 'n' variable and 'sizes[]'.
```suggestion
      int size = arrSizes[i], searchElement = expectedIndex[i][0], k = 0;
```